src/theory/quantifiers/sygus/sygus_pbe.cpp

```diff
@@ -112,17 +134,22 @@ void SygusPbe::collectExamples(Node n,
     for( unsigned i=0; i<n.getNumChildren(); i++ ){
       bool newHasPol;
       bool newPol;
-      QuantPhaseReq::getPolarity( n, i, hasPol, pol, newHasPol, newPol );
-      collectExamples( n[i], visited, newHasPol, newPol );
+      QuantPhaseReq::getEntailPolarity(n, i, hasPol, pol, newHasPol, newPol);
+      if (!collectExamples(n[i], visited, newHasPol, newPol))
```

Comment:
Yes, the PR fixes two bugs in this function. (1) pol was negated throughout this call, which is incompatible with getEntailPolarity, which (2) should have been used throughout instead of getPolarity.

To give an example of the bad behavior, previously the conjecture ( P => f(0)=1) ^ ( Q => f(0)=3) could have been inferred to be an infeasible PBE conjecture since the two pairs involving f have constant polarity. However they don't have "entailed" polarity since P and Q can be false.

Reply:
Sounds good, thanks for the explanation.